Tensile/Hardware.py

```diff
@@ -53,30 +60,38 @@ def __lt__(self, other):
             myProcPred = next(iter(x for x in myAndPred.value if x.tag == "Processor"), None)
             myCUPred = next(iter(x for x in myAndPred.value if x.tag == "CUCount"), None)
             myCUCount = myCUPred.value if myCUPred != None else 0
+            myIsAPUPred = next(iter(x for x in myAndPred.value if x.tag == "IsAPU"), None)
+            myIsAPU = myIsAPUPred.value if myIsAPUPred != None else -1
         else:
             myProcPred = self.value
             myCUCount = 0
+            myIsAPU = -1
 
         if other.value.tag == 'And':
             otherAndPred = other.value
             otherProcPred = next(iter(x for x in otherAndPred.value if x.tag == "Processor"), None)
             otherCUPred = next(iter(x for x in otherAndPred.value if x.tag == "CUCount"), None)
             otherCUCount = otherCUPred.value if otherCUPred != None else 0
+            otherIsAPUPred = next(iter(x for x in otherAndPred.value if x.tag == "IsAPU"), None)
+            otherIsAPU = otherIsAPUPred.value if otherIsAPUPred != None else -1
         else:
             otherProcPred = other.value
             otherCUCount = 0
+            otherIsAPU = -1
 
-        # If CU properties are empty, then compare processor predicates
-        if myCUCount == otherCUCount == 0:
-            # Make sure that we have valid processor preds
-            assert myProcPred != None and otherProcPred != None, "Missing processor predicate"
-            assert myProcPred.tag == otherProcPred.tag == "Processor", "Invalid processor predicate"
+        if myIsAPU == otherIsAPU:
+            # If CU properties are empty, then compare processor predicates
+            if myCUCount == otherCUCount == 0:
+                # Make sure that we have valid processor preds
+                assert myProcPred != None and otherProcPred != None, "Missing processor predicate"
+                assert myProcPred.tag == otherProcPred.tag == "Processor", "Invalid processor predicate"
 
-            # Downgrade to base class so that we don't recurse
-            myProcPredCopy = copy.deepcopy(myProcPred)
-            otherProcPredCopy = copy.deepcopy(otherProcPred)
-            myProcPredCopy.__class__ = otherProcPredCopy.__class__ = Properties.Predicate
-            return myProcPredCopy < otherProcPredCopy
+                # Downgrade to base class so that we don't recurse
+                myProcPredCopy = copy.deepcopy(myProcPred)
+                otherProcPredCopy = copy.deepcopy(otherProcPred)
+                myProcPredCopy.__class__ = otherProcPredCopy.__class__ = Properties.Predicate
+                return myProcPredCopy < otherProcPredCopy
 
-        # Higher priority given to higher CU count
-        return myCUCount > otherCUCount
+            # Higher priority given to higher CU count
+            return myCUCount > otherCUCount
+        return myIsAPU > otherIsAPU
```

Comment:
Yes, if APU is set then that sorts first since it is a specialized tuning. Sort order is APU, XPU, generic. Updated the comments in this section with more detail.